docs/docs/concepts/infrastructure-architecture.mdx

```diff
@@ -0,0 +1,139 @@
+lifecycle, including authentication, spawning, and routing requests to the appropriate
+single-user server. The Hub also serves as the primary entry point for users, providing
+a web interface for logging in, selecting services, and launching single-user servers.
+
+Then there is the configurable http proxy, which acts as a gatekeeper, dynamically updating internal routing to direct traffic efficiently and securely to the appropriate server instance. This setup not only supports default proxies but also allows for custom proxy configurations to manage various external applications seamlessly.
+
+Last but not least, the Spawners are abstract interfaces to processes that can start,
+monitor, and stop single-user servers. They are responsible for managing the lifecycle
+of the user's server, including spawning and stopping the server, as well as monitoring
+its status.
+
+![JupyterHub’s Major subsystems](/img/jhub_subsystems.png)
+
+For more information on JupyterHub's architecture, please refer to the official
+[JupyterHub Technical Overview](https://jupyterhub.readthedocs.io/en/stable/reference/technical-overview.html#) documentation.
+
+## JHub Apps Integration
+
+When JHub Apps is launched, it is integrated into JupyterHub as a new service. This addition modifies the Hub's homepage to feature the service's URL, thereby offering users not only access to other services but also control over spawning different frameworks beyond JupyterLab, like Streamlit and Bokeh, while utilizing JupyterHub's authentication and spawner mechanisms.
+
+The newly added Jhub-apps service introduces intermediate steps that tailor request routing based on the selected framework:
+
+- The Hub initiates its own proxy server,
+- Jhub-apps is launched and registered within JupyterHub as an external service,
+- The proxy initially routes all requests to the Hub, while complying with the
+  extensions and modifications made by Jhub-apps,
+- Jhub-apps modifies the Hub's homepage to include its URL (dynamically handled by the
+  proxy above), as well as extending the service
+  selection options for users, who can now choose from a variety of frameworks.
+- The Hub continues to handle logins and server spawning,
+- Jhub-apps adjusts the request handling to direct users to the appropriate single-user
+  server environments based on their selections.
+
+## Technical Architecture Diagram
+
+Below is a diagram illustrating the technical architecture of the JHub Apps service and its interaction with the JupyterHub system:
+
+![JHub Apps Architecture](/img/jhub_service_diagram.png)
+
+Starting from the left of the diagram:
+- Users engage with the JupyterHub interface via their browsers, logging in through
+  their preferred OAuth providers (e.g., GitHub, Google). Authenticated users are
+  redirected to JHub's homepage, where they select and launch the desired service.
+- As an external server managed by JHub, the JHub Apps service maintains full access to
```

Comment:
external server -> external service
managed by JHub -> JupyterHub (I don't think its standard to use JHub for JupyterHub)